Assets/MixedRealityToolkit/Utilities/CameraCache.cs

```diff
@@ -34,9 +34,29 @@ public static Camera Main
                 var mainCamera = Camera.main;
 
                 if (mainCamera == null)
-                {   // If no main camera was found, create it now
-                    Debug.LogWarning("No main camera found. The Mixed Reality Toolkit requires at least one camera in the scene. One will be generated now.");
-                    mainCamera = new GameObject("Main Camera", typeof(Camera), typeof(AudioListener)) { tag = "MainCamera" }.GetComponent<Camera>();
+                {
+                    Debug.LogWarning("No main camera found. The Mixed Reality Toolkit requires one camera in the scene to be tagged as \"MainCamera\".");
+                    Debug.Log("Searching for cameras in the scene.");
+
+                    // If no main camera was found, try to determine one.
+                    Camera[] cameras = GameObject.FindObjectsOfType<Camera>();
+                    switch (cameras.Length)
+                    {
+                        case 0:
+                            Debug.LogWarning("No cameras found. Creating a \"MainCamera\".");
+                            mainCamera = new GameObject("Main Camera", typeof(Camera), typeof(AudioListener)) { tag = "MainCamera" }.GetComponent<Camera>();
+                            break;
+
+                        case 1:
+                            Debug.LogWarning("Camera found. Tagging the camera as \"MainCamera\".");
```

Comment:
This seems a little dangerous. Automatically tagging a camera on a developer that they intended for something else? And there is no way to turn this auto-functionality off

Reply:
MRTK will not function without a camera tagged as MainCamera. We can replace this with an error and require the developer to manually tag. Thoughts?

Reply:
Yea indeed, lots of null exceptions everywhere since we use CameraCache.Main everywhere.

I'm leaning to the FAIL fast approach and notify user of missing MainCamera tag vs silently create one. 

I know this will be rare but I could see some users trying to create new cameras based on what platform they are on or creating new cameras when changing scenes or something similar.

If I am just ignorant or screwed up and forgot to add a camera, I think most users will know how to tag a camera as main and fix the error. That is pretty common in Unity whereas a power user trying to do something hacky/unique, we will be blocking them from whatever they are trying to do

Reply:
actually , some of the test automation relies on MRTK creating a camera if one doesnt exist... i can fix these issues when i fix the automation. sound ok?